### PR TITLE
Make Runtime RID-agnostic and simplify infra

### PR DESCRIPTION
**Pull request by andrew-boyarshin**
* Reimplement RID-specific Runtime parts to be RID-independent
* Use proper stack trace method on .NET Framework, fallback is only for .NET Standard 1.1
* Use nightly Roslyn compiler for `nint` and `nuint`, used in new `NativeLong`/`NativeULong` implementation
* Fix wrong data model for Native LONG types (SharpGen was correct, Runtime was not, IntPtr is on Unix, int is on Windows, not the other way around)
* Fix bug in SharpGen MSBuild targets (it was impossible to override `SharpGenSdkAssembly` path)
* Improve MSBuild scripts
* Update dependencies
* Use centralized dependencies version management
* Make CI builds deterministic (still affected by current bug in .NET SDK)
* Improved test and coverage scripts
* Migrate to `net472;netcoreapp2.1` (MSBuild requires `net472;netcoreapp2.0`, .NET Core 2.0 is EOL, 2.1 is the closest LTS)
* Drop Refs since no RID-specific code is left in Runtime
* Fine-tune Runtime dependency graph
* Workaround MSBuild Core not loading `Microsoft.Win32.Registry` implementation dll (I suspect bug in MSBuild assembly load context override) by building RuntimeIdentifier-specific assemblies (`win;unix`)
* Drop SharpGen.Interactive

Note, that I'm not sure `nint` and `nuint` work on all .NET platforms supported by Runtime, it would be great to have this confirmed by someone more knowledgeable than me. Roslyn specs don't detail on that either. ~~Theoretically it should work, but I can't test it at the moment (still downloading Windows 8 RTM image 😄).~~ Works on .NET Framework 4.5 inbox on Windows 8 RTM x64.

Code Coverage is still here by default (`test.ps1`), but the results are empty on CI.

### CODE REVIEW

File: SharpGen.Runtime/PlatformDetection.cs [hunk 1]
```diff
@@ -1,20 +1,12 @@
...
-        public static bool IsWindows => true;
-        public static bool IsItaniumSystemV => false;
-#elif UNIX
-        public static bool IsWindows => false;
-        public static bool IsItaniumSystemV => true;
-#else
-        public static bool IsWindows => throw null;
-        public static bool IsItaniumSystemV => throw null;
-#endif
+        public static bool IsWindows => RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
```
Comment (jkoritzinsky) [inline SharpGen.Runtime/PlatformDetection.cs:7]: These implementations aren't JIT time constants. Can we calculate these in a static constructor and cache them in a bool field? That will make these more likely to be JIT time constants and get back the perf benefits from the rid-specific implementation.

Comment (andrew-boyarshin) [reply SharpGen.Runtime/PlatformDetection.cs:7]: Maybe turn them into `static readonly` fields instead of computed properties with backing fields?

Comment (jkoritzinsky) [reply SharpGen.Runtime/PlatformDetection.cs:7]: Yes lets do that.

**Comment by andrew-boyarshin** [comment]
@jkoritzinsky I've confirmed that this works correctly on Windows 8 RTM (6.2.9200 x64, .NET Framework 4.5, 0 installed updates from Windows Update).

Found issues with many `SdkTests` not working on x86 (environment doesn't matter). Since the issue is present even before this PR, it's not related to changes here.

**Comment by andrew-boyarshin** [comment]
@jkoritzinsky is there anything else you want done before merging this PR?

**Comment by jkoritzinsky** [comment]
Nope! I just have been really busy and haven’t had a chance to merge this in.

**Comment by jkoritzinsky** [comment]
Actually, can you fix the CodeFactor check?

**Comment by andrew-boyarshin** [comment]
@jkoritzinsky I'm not proficient with PowerShell (I don't know correct `Write-Host` replacement). The issues are from the code I've moved, but not modified.

**Comment by jkoritzinsky** [comment]
Ok those are fine. I’ll merge this in tomorrow when I’m at my computer.

**Pull request merged**